### DISCUSSION TITLE
Compare / Diff between two version of Compilation

### DISCUSSION THREAD

**Discussion by mert-kurttutan** [pacak/cargo-show-asm#181] [Q&A] [answered]
Hi,
Thanks for your project, it has been really useful. 
I have two compilation for the same project, one with target-cpu set to native and the one with not set to native.

I want to compare asm of two compilations if I am missing any target-specific inlining since I am doing run-time detection of target cpu.

At the moment, I compare individual functions, which becomes cumbersome as one function can make to another function, function list kind of becomes bigger.

Do you have any suggestion to make such a comparison process more automated instead of checking each function indvidually?

**Comment by mert-kurttutan** [comment]
Btw, I already use `--native` option

**Comment by pacak** [comment, marked as answer]
Hmm... There's `--everything` that dumps, well, everything. Also you can look at the name list itself - if function is inlined everywhere - it should not show up in the list.

**Comment by mert-kurttutan** [comment]
`--everything` seems to solve my problem. Thanks. I am closing the issue.

I am wondering, are you thinking of doing some IDE extension/tool based on this project?
Would it make sense to have such a tool?

**Comment by pacak** [comment]
> I am wondering, are you thinking of doing some IDE extension/tool based on this project?
> Would it make sense to have such a tool?

There's several problems - lots of different IDEs use all sorts of languages for their extension so supporting each separate IDE means doing more work. Next IDEs usually operate in terms of rust code directly, while `cargo-show-asm` talks about symbols found in generated binaries. While I can try to replicate this mapping - it will be chasing a moving target - symbols can change, what happens when function is inlined or optimized away, etc.

In short while I would love to make it easier to see compiled code directly from my editor - it's way too much effort compared to typing a few commands in a console.
